pkg/daemon/daemon.go

```diff
@@ -62,17 +65,21 @@ func New(
 		return nil, fmt.Errorf("Error establishing connection to logind dbus: %v", err)
 	}
 
-	if err := loadNodeAnnotations(kubeClient.CoreV1().Nodes(), nodeName); err != nil {
+	if err = loadNodeAnnotations(kubeClient.CoreV1().Nodes(), nodeName); err != nil {
 		return nil, err
 	}
 
+	osImageURL, osVersion, err := getBootedOSImageURL(rootMount)
```

Comment:
We may want to start denoting functions that are meant to run pre-chroot.

Reply:
True. I don't see a reason for my comment to hold this PR up though. It was more of a thought as I was reading through changes.